File: backend/app/routers/prescriptions.py
from fastapi import APIRouter, Depends, status
from sqlalchemy.ext.asyncio import AsyncSession
from app.dependencies.db import get_db
from app.dependencies.auth import get_current_user, require_role
from app.models.user import User, Role
from app.schemas.prescription import PrescriptionCreate, PrescriptionStatusUpdate, PrescriptionRead
from app.services import prescription_service

router = APIRouter()


@router.post("", response_model=PrescriptionRead, status_code=status.HTTP_201_CREATED)
async def create_prescription(
    data: PrescriptionCreate,
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(require_role(Role.doctor)),
):
    return await prescription_service.create_prescription(db, data, current_user.id)


@router.get("/active", response_model=list[PrescriptionRead])
async def get_active_prescriptions(
    db: AsyncSession = Depends(get_db),
    _: User = Depends(require_role(Role.doctor, Role.pharmacist, Role.admin)),
):
    return await prescription_service.get_active_prescriptions(db)


@router.patch("/{prescription_id}/status", response_model=PrescriptionRead)
async def update_prescription_status(
    prescription_id: str,
    data: PrescriptionStatusUpdate,
    db: AsyncSession = Depends(get_db),
    _: User = Depends(require_role(Role.doctor, Role.pharmacist, Role.admin)),
):
    return await prescription_service.update_prescription_status(db, prescription_id, data.status)

File: frontend/src/api/prescriptions.ts
import api from './axios'

export interface Prescription {
  id: string
  patient_id: string
  prescribed_by: string
  medication_name: string
  dosage?: string
  instructions?: string
  status: 'Pending' | 'Approved' | 'Dispensed'
  created_at: string
  updated_at: string
}

export const createPrescription = (data: {
  patient_id: string
  medication_name: string
  dosage?: string
  instructions?: string
}) => api.post<Prescription>('/prescriptions', data).then((r) => r.data)

export const getActivePrescriptions = () =>
  api.get<Prescription[]>('/prescriptions/active').then((r) => r.data)

export const updatePrescriptionStatus = (id: string, status: string) =>
  api.patch<Prescription>(`/prescriptions/${id}/status`, { status }).then((r) => r.data)
